### PR TITLE
Wait for onSdkReady callback to call isVerified

### PR DESCRIPTION
**Pull request by joelamouche**
**JIRA task url** (if applicable)
https://allianceblockprotocol.atlassian.net/browse/NEXID-1732

**What kind of change does this PR introduce?** (Bug fix, feature, docs update, ...)
- wait for onSdkReady to call isWhitelisted
- button is in loading status in the meantime
- isWhitelisted is not called in the begining

**How to reproduce fix / test this PR?**

### CODE REVIEW

File: aave_interface/src/modules/dashboard/lists/SuppliedPositionsList/SuppliedPositionsListItem.tsx [hunk 1]
```diff
@@ -130,7 +130,9 @@ export const SuppliedPositionsListItem = ({
             onClick={handleListButton}
           >
             {isAuthenticated ? (
-              isWhitelisted ? (
+              whitelistStatusLoading ? (
+                <Trans>Loading Whitelist...</Trans>
```
Comment (LizanLycan) [inline aave_interface/src/modules/dashboard/lists/SuppliedPositionsList/SuppliedPositionsListItem.tsx:134]: I think that we could add this loading state to the `disabled` logic of the button

File: aave_interface/src/modules/dashboard/lists/SupplyAssetsList/SupplyAssetsListItem.tsx [hunk 2]
```diff
@@ -108,7 +108,9 @@ export const SupplyAssetsListItem = ({
           onClick={handleListButton}
         >
           {isAuthenticated ? (
-            isWhitelisted ? (
+            whitelistStatusLoading ? (
+              <Trans>Loading Whitelist...</Trans>
```
Comment (LizanLycan) [inline aave_interface/src/modules/dashboard/lists/SupplyAssetsList/SupplyAssetsListItem.tsx:112]: I think that we could add this loading state to the `disabled` logic of the button

File: aave_interface/src/modules/dashboard/lists/SupplyAssetsList/SupplyAssetsListMobileItem.tsx [hunk 3]
```diff
@@ -115,7 +115,9 @@ export const SupplyAssetsListMobileItem = ({
           fullWidth
         >
           {isAuthenticated ? (
-            isWhitelisted ? (
+            whitelistStatusLoading ? (
+              <Trans>Loading Whitelist...</Trans>
```
Comment (LizanLycan) [inline aave_interface/src/modules/dashboard/lists/SupplyAssetsList/SupplyAssetsListMobileItem.tsx:119]: I think that we could add this loading state to the `disabled` logic of the button

**Review by LizanLycan — COMMENTED** [review]
Just a little bit comments

File: aave_interface/src/modules/dashboard/lists/SuppliedPositionsList/SuppliedPositionsListItem.tsx (hunk 1 above)
Comment (joelamouche) [reply aave_interface/src/modules/dashboard/lists/SuppliedPositionsList/SuppliedPositionsListItem.tsx:134]: yes good call!
I added it

File: aave_interface/src/modules/dashboard/lists/SupplyAssetsList/SupplyAssetsListMobileItem.tsx [hunk 4]
```diff
@@ -114,15 +120,13 @@ export const SupplyAssetsListMobileItem = ({
...
-          {isAuthenticated ? (
-            isWhitelisted ? (
-              <Trans>Supply</Trans>
-            ) : (
-              <Trans>Not whitelisted</Trans>
-            )
-          ) : (
-            <Trans>Need Auth</Trans>
-          )}
+          {SupplyButton({
```
Comment (LizanLycan) [inline aave_interface/src/modules/dashboard/lists/SupplyAssetsList/SupplyAssetsListMobileItem.tsx:123]: Ok... I understand the intention, but it's a React component, please call it as a React component:
`<SupplyButton disableSupply={disableSupply} ... />`

File: aave_interface/src/modules/dashboard/lists/SupplyButton.tsx [hunk 5]
```diff
@@ -0,0 +1,30 @@
...
+import { Button } from '@mui/material';
+
+type SupplyButtonInputs = {
+  disableSupply: boolean;
+  isAuthenticated: boolean;
+  whitelistStatusLoading: boolean;
+  isWhitelisted: boolean;
+  handleListButton: () => void;
+};
+export const SupplyButton = (inputs: SupplyButtonInputs) => (
```
Comment (LizanLycan) [inline aave_interface/src/modules/dashboard/lists/SupplyButton.tsx:11]: Generally for a React Component we call the `inputs` `props`

**Review by LizanLycan — CHANGES_REQUESTED** [review]

File: aave_interface/src/modules/dashboard/lists/SupplyButton.tsx (hunk 5 above)
Comment (joelamouche) [reply aave_interface/src/modules/dashboard/lists/SupplyButton.tsx]: Good point, corrected

**Review by LizanLycan — APPROVED** [review]
LGTM

**Pull request merged**